### PR TITLE
Added run criteria example at examples/ecs.

### PR DESCRIPTION
**Pull request by TuNiman1**
# Objective

- I added an example for run criteria currently located at examples/ecs/run_criteria.rs
- This is a simple example for beginners explaining and present how run criterias in bevy works.

## Changelog

> Added my change in the changelog.

- Basically, i helped the beginners of the community by programming an example for run criteria.
- I put my changes in the Added category in the CHANGELOG.md file
  - This example would help beginners in this amazing library to use run criteria and continue their journey in the library.

### CODE REVIEW

File: examples/ecs/run_criteria.rs
Comment (alice-i-cecile) [inline examples/ecs/run_criteria.rs:1]: This should use //!

Comment (alice-i-cecile) [inline examples/ecs/run_criteria.rs:1]: Can you add more color here, explaining what run criteria are and why they're useful?

Comment (TuNiman1) [reply examples/ecs/run_criteria.rs:1]: yeah sure, sorry that i forgot to

File: examples/ecs/run_criteria.rs
Comment (alice-i-cecile) [inline examples/ecs/run_criteria.rs:43]: Doesn't this currently fail?

Comment (TuNiman1) [reply examples/ecs/run_criteria.rs:43]: nope, checked it many times

Comment (TuNiman1) [reply examples/ecs/run_criteria.rs:43]: currently ive checked it on linux and windows

**Comment by alice-i-cecile** [comment]
I appreciate it, and think this should exist. This is blocked on #7267 though, which is nearly complete, and changes how run criteria (now run conditions) work.

File: CHANGELOG.md
Comment (alice-i-cecile) [inline CHANGELOG.md:16]: This semi-automatically generated each release; can you revert your changes to this file?

Comment (TuNiman1) [reply CHANGELOG.md:16]: wait a second

**Comment by TuNiman1** [comment]
wait what do i need to do now? add more information about what run criteria is?

**Comment by TuNiman1** [comment]
i don't get it, why does it failed on the check-missing-examples-in-docs

**Comment by alice-i-cecile** [comment]
> i don't get it, why does it failed on the check-missing-examples-in-docs

Like it says in the job (you can click through), there's a build script that needs to be run to synchronize the contents of our Cargo.toml and the examples/README.md.

File: examples/ecs/run_criteria.rs
Comment (alice-i-cecile) [inline examples/ecs/run_criteria.rs:3]: Stick to //! comments here, for consistency with the other examples

File: examples/ecs/run_criteria.rs
Comment (alice-i-cecile) [inline examples/ecs/run_criteria.rs:1]:
```suggestion
//! Demonstrates how and why to use run criteria to control whether or not systems run
```

**Comment by TuNiman1** [comment]
by the way, thank you for all your help, this really not obvious

**Comment by TuNiman1** [comment]
why is it still failed
i already added the example metadata in the cargo.toml file

**Comment by TuNiman1** [comment]
maybe this will do

**Comment by alice-i-cecile** [comment]
> by the way, thank you for all your help, this really not obvious

No worries, learning how to deal with the CI and our processes can take a while :)

**Comment by TuNiman1** [comment]
> 

Oh i didnt saw it sorry

**Comment by TuNiman1** [comment]
where is the build script located at?

File: examples/ecs/run_criteria.rs
Comment (alice-i-cecile) [inline examples/ecs/run_criteria.rs:4]: This paragraph could use a bit more clarity, and a more formal tone :)

I would try to cover:

1. What a run criteria is / what it does
2. Some practical uses
3. Any restrictions on what types of functions can be used as run criteria

**Comment by TuNiman1** [comment]
whatever it worked

File: examples/ecs/run_criteria.rs
Comment (TuNiman1) [reply examples/ecs/run_criteria.rs:4]: Alright

File: examples/ecs/run_criteria.rs
Comment (alice-i-cecile) [inline examples/ecs/run_criteria.rs:46]:
```suggestion
                // You can combine run criteria
                // In this case, the system controlled by this run criteria is only evaluated a single time
```

Comment (TuNiman1) [reply examples/ecs/run_criteria.rs:46]: yeah forgot to change it, i typed its major cause otherwise it will run an infinite look on the text.

**Comment by TuNiman1** [comment]
what happend this time come on

**Comment by TuNiman1** [comment]
why is it still failing oh my god

**Comment by TuNiman1** [comment]
FINNALY, im so happy

**Comment by TuNiman1** [comment]
So, what do i do now? when will you merge it?

**Comment by alice-i-cecile** [comment]
> So, what do i do now? when will you merge it?

In general, we need 2 approvals before a PR can be merged :) Ask around on Discord if it's been a while without feedback.

In this case though, like I said above, this PR can't / shouldn't be merged until #7267 is merged, which will force some changes to it to reflect the new API :)

**Comment by djeedai** [comment]
@TuNiman1 can you please edit the changelog section of PR description to contain what needs to be copied into the future changelog, and not a description of where you put your changelog things? I _think_ we have release scripts running that will copy that part into the changelog, so currently it will copy the entire block of text including the boilerplate not deleted and your explanations, but will not contain the actual changelog content which is missing. So instead it should be a one-liner like:

```
## Changelog

- Added a new `run_criteria` example demonstrating run criterias.
```

**Review by djeedai — CHANGES_REQUESTED** [review]
I believe the changelog section of the PR description is not in the right format.

**Comment by TuNiman1** [comment]
Yeah sure

**Comment by TuNiman1** [comment]
done @djeedai

**Comment by TuNiman1** [comment]
hope it's what you wanted

**Comment by TuNiman1** [comment]
although it all runs just fine

**Comment by JoJoJet** [comment]
#7267 has been merged, so this should be un-blocked now. This PR will need to be updated to account for the fact that run criteria have been replaced with run conditions :)

**Comment by TuNiman1** [comment]
So, what now? Do I need to do something at the moment?

**Comment by TuNiman1** [comment]
I'll maybe add one more commit on something that currently does nothing in the code

**Comment by alice-i-cecile** [comment]
> So, what now? Do I need to do something at the moment?

So you'll want to rebase this PR to the latest `main` in order to grab the new changes :) This will resolve the merge conflicts.

Once that's working, I'd be happy to merge this. Renaming the PR title would be nice too; they're now "run conditions".

**Comment by TuNiman1** [comment]
Ohhhhh like update it the the newest version?

**Comment by alice-i-cecile** [comment]
Yep, precisely :)

**Comment by alice-i-cecile** [comment]
@TuNiman1 are you able to tackle this? Totally fine if not: I'll just add the `Adopt-Me` label and someone else will pick this up and keep your credit for this work.

**Comment by TuNiman1** [comment]
Oh! Thanks for reminding me! I will do it in about an hour

**Comment by TuNiman1** [comment]
But I will tackle this, I prefer not to adopt to others

**Comment by TuNiman1** [comment]
Again, sorry for another stupid question but as you know it is my first time contributing and do I need to just do a simple git pull to tackle this?

**Comment by TuNiman1** [comment]
And my English is not native but very good for a 13 year old, still not close to native so if I don't understand something I'll probably ask it again and im sorry for that lol

**Comment by alice-i-cecile** [comment]
Yep, not a problem, your English is exceptional :D  To fix the merge conflicts, there are two options:

1. Merge `main` into your branch and resolve any merge conflicts. 
2. Rebase your branch onto `main`, resolving any conflicts at each commit.
3. Use Github's built in "Resolve conflicts" button on this PR.

Any of them is fine here: approach 1 preserves history but results in a big merge commit, approach 2 is "cleaner" but rewrites history. Approach 3 is great for very simple conflicts.

In my experience, tools like GitKraken can be very helpful for navigating this sort of more advanced workflow :)

**Comment by TuNiman1** [comment]
One more question, when I do it will you merge my request?

**Comment by alice-i-cecile** [comment]
After we get approvals from me + one other person, yes :) I'd love to get this out for 0.10 at the end of the week to help new users learn, so I'll be recruiting reviewers agressively.

**Comment by TuNiman1** [comment]
So glad to hear that!

**Comment by TuNiman1** [comment]
Lol that was scary easy

File: CHANGELOG.md
Comment (alice-i-cecile) [inline CHANGELOG.md:16]: Can you revert the changes to `CHANGELOG.md`? This tends to be generated all in one batch at each release.

Comment (TuNiman1) [reply CHANGELOG.md:16]: indeed

**Comment by alice-i-cecile** [comment]
Taking a quick look at `main`, it appears that we've already added [an example for this](https://github.com/bevyengine/bevy/blob/main/examples/ecs/run_conditions.rs) in #7652 <3

Sorry for not catching that earlier! I'm going to close this out, but feel free to improve on that example in a new PR or make another: the help is much appreciated and we're always happy to help folks learn and improve.

**Comment by TuNiman1** [comment]
Although there, the example shows run if, not with run criteria, they are differ because run criteria stands for a system set and run if stands for one system I completely understand and I had a lot of fun trying to contribute and hopefully next time I will do great(:

**Comment by TuNiman1** [comment]
Upgrading the taffy dependency sound like a pretty good change! So I will

**Pull request closed without merging**
